src/manifolds/GraphManifold.jl

```diff
@@ -0,0 +1,89 @@
+
+using LightGraphs
+import LightGraphs: AbstractSimpleGraph
+
+@doc doc"""
+    GraphManifoldType
+
+This type represents the type of data on the graph that the [`GraphManifold`](@ref)
+represents.
+"""
+abstract type GraphManifoldType end
+
+@doc doc"""
+    EdgeGraphManifold <: GraphManifoldType
+
+A type for a [`GraphManifold`](@ref) where the data is given on the edges.
+"""
+struct EdgeGraph <: GraphManifoldType end
+
+@doc doc"""
+    VectexGraphManifold <: GraphManifoldType
+
+A type for a [`GraphManifold`](@ref) where the data is given on the vertices.
+"""
+struct VertexGraph <: GraphManifoldType end
+
+@doc doc"""
+    GraphManifold{G,M,T}
+
+Build a manifold, that is basically a [`PowerManifold`](@ref) either on the
+edges or vertices of a graph.
+
+# Fields
+* `G` is an `AbstractSimpleGraph`
+* `M` is a [`Manifold`](@ref)
+* `T` is a [`GraphManifoldType`](@ref), i.e. whether data is given on vertices
+or edges.
+"""
+struct GraphManifold{
+        G<:AbstractSimpleGraph,
+        M<:Manifold,
```

Comment:
You could constrain it to `M<:PowerManifold` I think.

Reply:
Neat idea, would spare quite a few definitions.